build/windows/wix_installer/sync-gateway.wxs

```diff
@@ -97,6 +97,10 @@
                    Type="ownProcess"
                    Arguments='start "[INSTALLDIR]$(var.WinSGWEXE)" "[INSTALLDIR]$(var.WinServiceConfig)" "[INSTALLDIR]var\lib\couchbase\logs\sync_gateway_error.log" '
                    Vital="yes">
+            <util:ServiceConfig
+                     FirstFailureActionType="restart"
+                     SecondFailureActionType="restart"
+                     ThirdFailureActionType="restart" />
```

Comment:
With the last change to make all 3 of these `"restart"` - does that mean the service indefinitely restarts if misconfigured?
Does that match the Linux services? I had a feeling they stopped after a certain number of attempts.

Reply:
It means the service will indefinitely restart. The service was already indefinitely restarting when only the first failure was set to restart and the rest were set to not take action on failure because the reset fail count was set to 0.

It seems the Linux services also don't stop after a certain number of attempts however if Sync Gateway fails too quickly (EG. incorrect JSON in the startup config), Systemd will not try to restart it. 

Windows does not implement this though so it just constantly retries, even if Sync Gateway stops instantly. 
This could be combatted in 2 ways:
- Have SGW only restart the first and second time when they fail, then reset daily.
- Use the service wrapper to handle SGW fails, then add logic in to that that copies Systemd's logic of not restarting the service if it fails too fast.

I am leaning towards the second option to prevent spam, and give us more control on what to do if SGW fails.

Reply:
>It seems the Linux services also don't stop after a certain number of attempts however if Sync Gateway fails too quickly (EG. incorrect JSON in the startup config), Systemd will not try to restart it.

Ahh ok, yeah that burst behaviour is definitely what I was thinking of!

In that case, I think indefinite but configurable retries on Windows are an acceptable compromise. It should be immediately obvious that SG has not started correctly when somebody has misconfigured something.

I'd rather not add any more stuff in the windows service wrapper if it's non-standard behaviour on that platform.